src/Microsoft.Azure.ServiceBus/SessionHandlerOptions.cs

```diff
@@ -83,11 +83,15 @@ public int MaxConcurrentSessions
                 this.MaxConcurrentAcceptSessionCalls = Math.Min(value, 2 * Environment.ProcessorCount);
             }
         }
-
+        
         /// <summary>Gets or sets whether the autocomplete option of the session handler is enabled.</summary>
         /// <value>true if the autocomplete option of the session handler is enabled; otherwise, false.</value>
         public bool AutoComplete { get; set; }
 
+         /// <summary>Gets or sets a single session ID. This option is optional.</summary>
+         /// <remark>Note that the handler will only listen to this single session ID for its entire lifetime.</remark>
+        public string SessionId { get; set; }
+
```

Comment:
What happens when you set sessionId here and then set maxConcurrentSessions as well?
The other args do not hold up when you put a sessionId here.

Reply:
Good question. I've missed that... normally there's be a default (2000) or user specified number of concurrent pumps. When `SessionId` is specified (i.e. not null), it should be short circuited to one pump only imo.